sycl/doc/extensions/SPIRV/SPV_INTEL_float_controls2.asciidoc

```diff
@@ -0,0 +1,206 @@
+This extension adds new decorations that can be applied to a function to control floating-point computations inside the function.
+
+
+Extension Name
+--------------
+
+To use this extension within a SPIR-V module, the following
+*OpExtension* must be present in the module:
+
+----
+OpExtension "SPV_INTEL_float_controls2"
+----
+
+Modifications to the SPIR-V Specification, Version 1.5, Revision 3, Unified
+---------------------------------------------------------------------------
+
+Modify Section 3.6, Execution Mode, add the following rows to the Execution Mode table:
+
+--
+[cols="1,20,10,10",options="header",width = "80%"]
+|====
+  2+^| Execution Mode  | Extra Operands | Enabling Capabilities
+| 5620 | *RoundingModeRTPINTEL* +
+The default rounding mode for floating-point arithmetic and conversions instructions must be round to positive infinity.
+If an instruction is decorated with *FPRoundingMode* or defines a rounding mode in its description, that rounding mode is applied and *RoundingModeRTPINTEL* is ignored.
+Only affects instructions operating on a floating-point type whose component width is _Target Width_.
+
+_Target Width_ is an unsigned 32-bit integer.
+| <<Literal, 'Literal'>>  _Target Width_ | *RoundToInfinityINTEL*
+| 5621 | *RoundingModeRTNINTEL* +
+The default rounding mode for floating-point arithmetic and conversions instructions must be round to negative infinity.
+If an instruction is decorated with *FPRoundingMode* or defines a rounding mode in its description, that rounding mode is applied and *RoundingModeRTNINTEL* is ignored.
+Only affects instructions operating on a floating-point type whose component width is _Target Width_.
+
+_Target Width_ is an unsigned 32-bit integer.
+| <<Literal, 'Literal'>> _Target Width_ | *RoundToInfinityINTEL*
+  | 5622 | *FloatingPointModeALTINTEL* +
+The default floating-point operation mode for floating-point arithmetic and conversions instructions must be the Alternative Floating-Point Mode as specified by the client API.
+Only affects instructions operating on a floating-point type whose component width is _Target Width_.
+
+_Target Width_ is an unsigned 32-bit integer.
+| <<Literal, 'Literal'>> _Target Width_ | *FloatingPointModeINTEL*
+| 5623 | *FloatingPointModeIEEEINTEL* +
+The default floating-point operation mode for floating-point arithmetic and conversions instructions must be the IEEE 754 Mode as specified by the client API.
+Only affects instructions operating on a floating-point type whose component width is _Target Width_.
+
+_Target Width_ is an unsigned 32-bit integer.
+| <<Literal, 'Literal'>> _Target Width_ | *FloatingPointModeINTEL*
+|====
+--
+Modify Section 3 Binary form, add new sub-sections after 3.16 FP Rounding Mode:
+--
+[[FP_Denorm_Mode]]*3.17. FP Denorm Mode*
+
+Denormalized values handling mode.
+
+[cols="^.^1,15,15",options="header",width = "50%"]
+|====
+2+^.^| FP Denorm Mode | Enabling Capabilities
+| 0x0 | *Preserve* +
```

Comment:
Is there is a reason to make it a bit enum instead of value enum? I don't think, we can apply both, so suggesting to change 0x0 and 0x1 to just 0 and 1.

Reply:
fixed